packages/taco/src/conditions/context/context.ts

```diff
@@ -112,7 +112,7 @@ export class ConditionContext {
   }
 
   private isContextParameter(param: unknown): boolean {
-    return typeof param === 'string' && param.startsWith(CONTEXT_PARAM_PREFIX);
+    return String(param).startsWith(CONTEXT_PARAM_PREFIX);
```

Comment:
How about matching with the regexp instead?

Reply:
Well, I think I'm going to remove that regexp all together, for consistency with `nucypher`. This is how a similar method is implemented in `nucypher`:
```
def is_context_variable(variable) -> bool:
    return isinstance(variable, str) and variable.startswith(_CONTEXT_PREFIX)
```

Reply:
> Why wouldn't we want :1234 as a condition parameter? Or :1abc? These parameters are currently accepted by nucypher (...)
> Well, I think I'm going to remove that regexp altogether, for consistency with nucypher

I think you should bring the regex back. I just wrote a test that shows that nucypher accepts all kinds of crazy context variable names:
```py
    a_context_variable_name = ":::::asdf//asdf         asdf"
    another_context_variable_name = ":12 \" ¿? \n "
```
For the sake of UX, we should keep consistency with variable names in regular coding languages (i.e., only letters, numbers and underscores, usually not starting with a number). I'm opening a PR in nucypher shortly to make it consistent with this.

Reply:
I want to maintain consistency between `nucypher-ts` and `nucypher` to avoid weird issues down the road (the former should be downstream of the latter). But if we could address this in both places I think that would be great.

Reply:
> I'm opening a PR in nucypher shortly to make it consistent with this.

@cygnusv once you open the PR do you mind if I piggyback a couple of small robustness improvements to that PR?

Reply:
@cygnusv you are welcome to add any commits to https://github.com/nucypher/nucypher/pull/3330. Up to you.

Reply:
Opened https://github.com/nucypher/nucypher/pull/3331. @piotr-roslaniec see that I used a more restricted regex since I suspect that `\w` behavior may different between TS and Python. In any case, the difference lies in accepting unicode characters, which I don't think it's very necessary anyway, but I'm open to suggestions (as long as we can keep complete consistency between TS and Python)

Reply:
I will update this PR in accordance with the draft here: https://github.com/nucypher/nucypher/pull/3331